File: site/src/types/recommendation.ts
export type ConfidenceLevel = 'STRONG' | 'MODERATE' | 'WEAK';

export type RecommendationType =
  | 'BTTS'
  | 'OVER_GOALS'
  | 'UNDER_GOALS'
  | 'BOOKING_POINTS'
  | 'VALUE_BET'
  | 'WINNING_FORM_MISMATCH'
  | 'LOSING_FORM_MISMATCH'
  | 'OVER_CORNERS'
  | 'UNDER_CORNERS'
  | 'CLEAN_SHEET'
  | 'FIRST_HALF_GOALS'
  | 'SECOND_HALF_GOALS'
  | 'MATCH_RESULT'
  | 'HOME_AWAY_SPECIALIST'
  | 'DRAW';

export interface Recommendation {
  fixtureId: number;
  homeTeamId: number;
  awayTeamId: number;
  homeTeamName: string;
  awayTeamName: string;
  matchDateUnix: number;
  leagueId: number | null;
  leagueName: string | null;
  leagueImage: string | null;
  type: RecommendationType;
  confidence: ConfidenceLevel;
  score: number;
  market: string;
  description: string;
  factors: Record<string, unknown>;
  generatedAt: string;
}

export interface RecommendationSummary {
  fixturesAnalyzed: number;
  totalRecommendations: number;
  strongRecommendations: number;
  moderateRecommendations: number;
  recommendationsByType: Record<RecommendationType, number>;
  generatedAt: string;
}

File: core/src/main/java/com/jcm/recommendations/soccer/core/recommendation/model/Recommendation.java
package com.jcm.recommendations.soccer.core.recommendation.model;

import lombok.Builder;
import lombok.Data;

import java.time.Instant;
import java.util.Map;

@Data
@Builder
public class Recommendation {
    
    private Long fixtureId;
    private Long homeTeamId;
    private Long awayTeamId;
    private String homeTeamName;
    private String awayTeamName;
    private Long matchDateUnix;
    
    private Long leagueId;
    private String leagueName;
    private String leagueImage;
    
    private RecommendationType type;
    private ConfidenceLevel confidence;
    private double score;
    private String market;
    private Double odds;
    private String description;
    
    private Map<String, Object> factors;
    
    private Instant generatedAt;
    
    public boolean isStrong() {
        return confidence == ConfidenceLevel.STRONG;
    }
    
    public boolean isModerate() {
        return confidence == ConfidenceLevel.MODERATE;
    }
    
    public boolean isActionable() {
        return confidence != ConfidenceLevel.WEAK;
    }
}
